main.js

```diff
@@ -386,9 +392,22 @@ $(function() {
       showMessage('0 units in town - can not buy.');
       return;
     }
-    inventory[productIndex] += 1;
-    towns[currentTown].stockpiles[productIndex] -= 1;
-    money -= price;
+
+    //if player selected "buy all", count is set to highest purchesable amount
+    const all = towns[currentTown].stockpiles[productIndex];
```

Comment:
Using the variable name `all` and the string `"all"` is quite confusing. This variable should be renamed to something like `itemsAvailable` or `countAvailable`.